core/storage/question/gae_models.py

```diff
@@ -354,15 +354,26 @@ def get_question_skill_links_based_on_difficulty_equidistributed_by_skill(
 
         question_skill_link_mapping = {}
 
+        def get_offset(query):
+            """Helper function to get the offset."""
+            question_count = query.count()
```

Comment:
From the documentation:
> count (read_policy=STRONG_CONSISTENCY, deadline=60, offset=0, limit=1000, start_cursor=None, end_cursor=None)
> Returns the number of results matching the query. This is faster by a constant factor than actually retrieving all of the results, but the running time still scales linearly with the sum of offset + limit. Unless the result count is expected to be small, it is best to specify a limit argument; otherwise the method will continue until it finishes counting or times out.

It seems like the running time of this method still scales linearly, which could be concerning. It is true that it's faster than retrieving all results. @seanlip Is this way good enough?

Reply:
I think it's probably OK. Let's go with it for now. But make sure there's a limit. Also do some manual testing to see what the performance times are and how they scale with different numbers of items in the datastore.

Reply:
Is there a particular way I can test, one way is to create the questions and fetch them in practice session but I am not sure how to time the event.

Reply:
Just noting, for posterity, that the thread is continued [here](https://github.com/oppia/oppia/pull/9061#issuecomment-629765809). (Sorry for breaking it off into the main thread -- github doesn't seem to have a way to "reply to thread" by email.)